### ISSUE TITLE
Open street maps are blurry and have poor quality in the new UI compared to openstreetmaps.org and the old UI

### ISSUE THREAD

**Issue by pc500** [traccar/traccar-web#827]
**Describe the bug**
Open street maps are blurry and have poor quality in the new UI compared to openstreetmaps.org and the old UI

**To Reproduce**
Open a map in the new UI + old UI as shown below, on a 1080p monitor with no DPI scaling (100%) enabled on Windows.

![image](https://user-images.githubusercontent.com/33823824/111091728-28b70580-84f1-11eb-8616-3543ef3423bf.png)

**Expected behavior**
A clearer map as shown in the old UI

**Screenshots**

**Desktop (please complete the following information):**
Windows 10, latest auto-updated Chrome (as of this writing).

**Smartphone (please complete the following information):**

**Additional context**
See https://www.traccar.org/forums/topic/open-street-maps-are-blurry-and-have-poor-quality-in-the-new-ui-compared-to-openstreetmapsorg/#post-60637 for more info.

**Comment by mtahasahin** [comment]
I looked into this bug and found that it is a result of two things:

1. OSM tile size is not specified in the code. MapboxGL defaults to 512 while it should be 256. Fixed in https://github.com/traccar/traccar-web/pull/829
2. Since it is a raster map the zoom level should always be an integer. Otherwise the map will be stretched and look blurry. This is not specific to OSM, any raster map will suffer from this.

I could think of two possible solutions. 
1. Allow only integer zoom levels for Map component. This will solve the problem for raster maps but also limit vector maps to integer zoom levels. (While vector maps can be displayed in arbitrary zoom levels because, well, they are vectors.) Easy to implement.
2. If a raster map style is selected allow only integer zoom levels. Otherwise allow any zoom level. This is a better solution but might add a little complexity to code.

Your thoughts? @tananaev
